src/View/Form/ArrayContext.php

```diff
@@ -194,16 +201,36 @@ public function val($field, $options = [])
      * @return bool
      */
     public function isRequired($field)
+    {
+        return (bool)$this->getRequiredMessage($field);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getRequiredMessage($field)
     {
         if (!is_array($this->_context['required'])) {
-            return false;
+            return null;
         }
         $required = Hash::get($this->_context['required'], $field);
         if ($required === null) {
             $required = Hash::get($this->_context['required'], $this->stripNesting($field));
         }
 
-        return (bool)$required;
+        if ($required === false) {
+            return null;
+        }
+
+        if ($required === true) {
+            $required = 'This field is required';
+
+            if (function_exists('__d ')) {
```

Comment:
Is this check necessary? We don't have a split package for view layer so won't `__d()` alway be available?

Reply:
Also you have an extra space at end of `'__d '` so the check would probably never return `true` :slightly_smiling_face:.

Reply:
That was pulled from the validation package which required the check since it's a standalone. But now that the logic is here I can remove the check. Not sure how the extra space got in there 😄